File: ejercicio_4.py
def palindromic():
    n = 0
    for i in range(100, 999):
        for j in range (999, 100, -1):
            cal = i * j
            if cal > n: 
                a = str(cal)
                if a[:] == a[::-1]: #[::-1] el string lo pongo al reves ej. '1234' -> '4321'
                    n = i * j
    print ("\nThe largest palindromic is: ", end = '')
    return n
print(palindromic())

File: ejercicio_4.cpp
#include <string>
#include <algorithm>
#include <iostream>

using namespace std;
int n = 0;
int cal = 0;
int j, i;
string t, a;

int main(){
string re;

	for(i = 100; i <= 999; i++){
		for(j = 999; j > 10; j--){
			//cout << i << "*"<< j << "\n";
			cal = i * j;
	  		if (cal > n){
				a = to_string(cal);
				t = a;
				reverse(a.begin(), a.end());
				if (t == a){
					n = i * j;
					//cout << i << "*" << j << "=" << n <<"\n"; 
				}
			}
		}
	}
	cout <<"The largest palindromic is: " << n <<"\n"; 
}